### PR TITLE
chore(build): use karma, saucelabs and travis for continuous integration

### PR DESCRIPTION
**Pull request by sadesyllas**

### CODE REVIEW

**Comment by esarbanis** [comment]
Here are some useful links that may help you setup saucelabs integration.

[Jasmine Testing Article](https://saucelabs.com/resources/articles/jasmine-testing)
[An example using grunt](https://github.com/axemclion/grunt-saucelabs)
[An example using gulp](https://github.com/pkozlowski-opensource/gulp-karma-sauce)

**Comment by vellerefond** [comment]
@esarbanis, thanks for the info but the problem is that saucelabs tests run fine on the branch. It's on the pull request that they fail and unfortunately, there are known issues with this process.

File: karma.conf.js [hunk 1]
```diff
@@ -1,3 +1,33 @@
+const fs = require('fs');
+
+// Use ENV vars on Travis and sauce.json locally to get credentials
+if (!process.env.TRAVIS && !process.env.SAUCE_USERNAME) {
+  if (!fs.existsSync('./sauce.json')) {
+    console.log('Please, create a sauce.json with your SauceLabs credentials.');
+    process.exit(1);
```
Comment (esarbanis) [inline karma.conf.js]: This means that it is required to have Saucelabs account in order to run the tests even locally.

Maybe we should just print a warning and go ahead and use Chrome as it is configured further down.

**Comment by esarbanis** [comment]
Reading Travis documentation, I bumped into [this](https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings).

What caught my eye was:
> Similarly, we do not provide these values to untrusted builds, triggered by pull requests from another repository.

**Comment by vellerefond** [comment]
Yes, these are the known issues :smiley:! As for saucelabs we have 2 accounts that we can use. One solution would be using firefox for pull requests since we have that info from travis' env (chrome is too old in travis to use) and then do a full test on a merged branch. Working on it...

**Comment by bekos** [comment]
@sadesyllas Awesome! Did some cleanup and landed as f2272d42b68bda15b074760a97df86e3dccbdbbe. Thx.

**Pull request closed without merging**